index.js

```diff
@@ -63,6 +64,21 @@ class ReadStream extends Readable {
       this.push(null)
       return cb(null)
     }
+    if (this.batch > 1) {
+      const batchStart = this.index
+      const batchEnd = Math.min(batchStart + this.batch, this.end, this.feed.length)
+      if (batchStart < batchEnd) {
```

Comment:
i don't think this would ever trigger an else? ie, can we drop the if

Reply:
There could be an "undownload" operation or something alike happening inbetween. But testing for that is surely tricky.

Reply:
I don't see how that would trigger it still

Reply:
Okay, I looked into it a bit more. I am not sure I am able to phrase this correctly, but you can replicate my findings by throwing an error statement in a else block: `} else { throw new Error('foo') }` and then run the tests (this case is actually already covered by the tests case.

My understanding, formatted in a way that I could add it as comment to the code :)

```js
// A batched live stream may start with an empty stream, resulting in both
// `batchEnd` and `batchStart` to be `0`. In this case a download needs to take place and
// we fall back to `get()` operation as it will trigger a download of the blocks linearly.
```

Reply:
this feels like it should be encapsulated with a IsBatchFinished or something that returns a bool, which you can call inside of if logic. Thank you for your contribution.